src/overlays/actors/ovl_En_Lift_Nuts/z_en_lift_nuts.c

```diff
@@ -112,109 +132,143 @@ static ColliderCylinderInit sCylinderInit = {
+    static s32 sAddedSharedMemory = false;
 
-    if (!D_80AEBF64) {
-        func_800B6584(play, ACTOR_EN_GAMELUPY, &D_80AEBF60, sizeof(s16));
-        D_80AEBF64 = true;
+    if (!sAddedSharedMemory) {
+        Actor_AddSharedMemoryEntry(play, ACTOR_EN_GAMELUPY, &sMinigameScore, sizeof(s16));
+        sAddedSharedMemory = true;
     }
-    this->ptr_1EC = &D_80AEBF60;
+    this->minigameScore = &sMinigameScore;
 }
 
-void func_80AE9A80(EnLiftNuts* this, PlayState* play) {
-    static s16 D_80AEBF68 = false;
+void EnLiftNuts_FreeSharedMemoryEntry(EnLiftNuts* this, PlayState* play) {
+    static s16 sFreedSharedMemory = false;
 
-    if (!D_80AEBF68) {
-        func_800B6608(play, ACTOR_EN_GAMELUPY);
-        D_80AEBF68 = true;
+    if (!sFreedSharedMemory) {
+        Actor_FreeSharedMemoryEntry(play, ACTOR_EN_GAMELUPY);
+        sFreedSharedMemory = true;
     }
 }
 
-s32 func_80AE9AC4(EnLiftNuts* this, s32 arg1) {
-    static s32 D_80AEBF6C = 0;
+typedef enum {
+    /* 0 */ ENLIFTNUTS_AUTOTALK_MODE_CHECK_OFF,
+    /* 1 */ ENLIFTNUTS_AUTOTALK_MODE_SET_ON,
+    /* 2 */ ENLIFTNUTS_AUTOTALK_MODE_SET_OFF
+} EnLiftNutsAutoTalkMode;
 
-    switch (arg1) {
-        case 0:
-            if (D_80AEBF6C == 0) {
+s32 EnLiftNuts_AutoTalk(EnLiftNuts* this, EnLiftNutsAutoTalkMode mode) {
+    static s32 sIsAutotalkOn = false;
+
+    switch (mode) {
+        case ENLIFTNUTS_AUTOTALK_MODE_CHECK_OFF:
+            if (!sIsAutotalkOn) {
                 return true;
             }
             break;
 
-        case 1:
-            if (D_80AEBF6C == 0) {
-                this->unk_34E = 1;
-                D_80AEBF6C = 1;
+        case ENLIFTNUTS_AUTOTALK_MODE_SET_ON:
+            if (!sIsAutotalkOn) {
+                this->autotalk = true;
+                sIsAutotalkOn = true;
                 return true;
             }
             break;
 
-        case 2:
-            if (D_80AEBF6C == 1) {
-                this->unk_34E = 0;
-                D_80AEBF6C = 0;
+        case ENLIFTNUTS_AUTOTALK_MODE_SET_OFF:
+            if (sIsAutotalkOn == true) {
+                this->autotalk = false;
+                sIsAutotalkOn = false;
                 return true;
             }
             break;
 
         default:
             break;
     }
+
     return false;
 }
 
-s32 func_80AE9B4C(s32 arg0, s32 arg1) {
-    static s32 D_80AEBF70 = 0;
+typedef enum {
+    /* 0 */ ENLIFTNUTS_MINIGAME_STATE_MODE_CHECK,
+    /* 1 */ ENLIFTNUTS_MINIGAME_STATE_MODE_SET
+} EnLiftNutsMiniGameStateMode;
+
+typedef enum {
+    /* 0 */ ENLIFTNUTS_MINIGAME_STATE_NONE,
+    /* 1 */ ENLIFTNUTS_MINIGAME_STATE_STARTING,
+    /* 2 */ ENLIFTNUTS_MINIGAME_STATE_RUNNING,
+    /* 3 */ ENLIFTNUTS_MINIGAME_STATE_AFTER,
+    /* 4 */ ENLIFTNUTS_MINIGAME_STATE_HIDING
+} EnLiftNutsMiniGameState;
+
+/**
+ * Will either check or set the current minigame state based on the given mode
+ */
+s32 EnLiftNuts_MinigameState(EnLiftNutsMiniGameStateMode mode, EnLiftNutsMiniGameState state) {
```

Comment:
This returns a boolean, but the name of the function doesn't immediately indicate what the boolean means in this context. Can you add to the comment what true/false from this function means?

Reply:
Added, let me know what you think.

Reply:
Looks good to me, thanks!